### ISSUE TITLE
`get_blob_properties` results in timeouts for non-existent blobs in version 0.15.0.preview

### ISSUE THREAD

**Issue by Gerg** [Azure/azure-storage-ruby#122]
We are using `azure-storage` by way of the `fog-azure-rm` gem. We noticed that when we upgraded from `0.14.0.preview` to `0.15.0.preview`, calls to `get_blob_properties` time out if the provided key does not exist.

Example:
```
service.get_blob_properties(directory.key, "key-that-does-not-exist", {})
=> Timeout: evaluation took longer than 10 seconds.
```

Downgrading back to `0.14.0.preview` resolved this issue. The version of `fog-azure-rm` that we are using depends on `azure-storage < 1.0`, so we are unable to upgrade beyond `0.15.0.preview` ([gemspec here](https://github.com/fog/fog-azure-rm/blob/4f8084621376dffb2342b8e09449eec28f18a221/fog-azure-rm.gemspec#L36)).

Thanks,
@gerg && @lisamcho

**Comment by yaxia** [comment]
Thanks for reporting the issue. I will take a look.

**Comment by yaxia** [comment]
@Gerg We have the fix in 1.1.0. Since it is GAed, we don't have a plan to release with version <1.0. Do you have any plan to upgrade to the latest? Thanks.

**Comment by yaxia** [comment]
Closing this issue since it is an issue in the old version.
